modules/renderer/features.js

```diff
@@ -94,6 +94,23 @@ export function rendererFeatures(context) {
         };
     }
 
+    defineRule('date_range', function isWithinRange(tags) {
+      // entity's start & end date
+      // filtering date range from the on-screen controls
+      const entityRange = {
+        'start_date': tags.start_date ? iD.utilNormalizeDateString(tags.start_date).value : iD.utilNormalizeDateString('-9999-12-31').value,
```

Comment:
I found that `utilNormalizeDateString()` can crash if `tags.start_date` or `tags.end_date` are _undefined_ or _null_ as opposed to an empty string. `utilNormalizeDateString()` goes straight into `var match = raw.match(dateRegex);` which fails if raw is undefined.

![image](https://user-images.githubusercontent.com/3117633/193683494-055e9ee2-96d5-480b-8c25-e31086770372.png)

I expect that adding `if (! raw) return null;` to `utilNormalizeDateString()` would work around this. That would handle blank, null, and undefined.

Reply:
Yes, feel free to make that change.

Reply:
I added the "if ! raw return null" to `utilNormalizeDateString()`

Then rewrote this piece of **features.js** to pick out normalized.value or else keep null, rather than specifying a fallback date.